lexers/c.lua

```diff
@@ -208,6 +214,31 @@ lex:set_word_list(lexer.ATTRIBUTE, {
 	'reproducible'
 })
 
+lex:set_word_list(lexer.TYPE .. '.linux', {
+	-- linux/types.h.
+	'u8', 'u16', 'u32', 'u64', 'u128', 's8', 's16', 's32', 's64', 's128',
+	'fd_set', 'dev_t', 'ino_t', 'mode_t', 'umode_t', 'nlink_t', 'off_t',
+	'pid_t', 'daddr_t', 'key_t', 'suseconds_t', 'timer_t', 'clockid_t',
+	'mqd_t', 'uid_t', 'gid_t', 'uid16_t', 'gid16_t', 'loff_t', 'caddr_t',
+	'aligned_u64', 'aligned_be64', 'aligned_le64', 'ktime_t', 'sector_t',
+	'blkcnt_t', 'pgoff_t', 'dma_addr_t', 'phys_addr_t', 'resource_size_t',
+	'irq_hw_number_t', 'atomic_t', 'atomic64_t', 'rcuref_t',
+	'rcu_head', 'rcu_callback_t', 'call_rcu_func_t', 'swap_r_func_t',
+	'swap_func_t', 'cmp_r_func_t', 'cmp_func_t'
+})
+
+lex:set_word_list(lexer.CONSTANT_BUILTIN .. '.linux', {
+	-- linux/limits.h.
+	'PHYS_ADDR_MAX', 'RESOURCE_SIZE_MAX', 'U8_MAX', 'S8_MAX', 'S8_MIN',
+	-- There is a #define U32_MIN ((u32)o), but not any of the others.
+	-- I'll refrain from going on a tirade here, but suffice to say it exists
+	-- to do codegen, but there don't appear to be any live references to it.
+	-- I'm going to see if it's really useless, and send a patch to remove
+	-- it if so. Feel free to not include it, as it may be gone fairly soon.
+	'U16_MAX', 'S16_MAX', 'S16_MIN', 'U32_MAX', 'U32_MIN',
+	'S32_MAX', 'S32_MIN', 'U64_MAX', 'S64_MAX', 'S64_min',
+})
+
```

Comment:
I would prefer not to have domain-specific things in a standard C lexer. See comment here: https://github.com/orbitalquark/scintillua/discussions/124#discussioncomment-11643726

Sorry about that.